server/api/routes.py

```diff
@@ -69,6 +70,9 @@ def login():
         return "Invalid username or password", 401
     # flask_login and this call to login_user handles the session for us
     login_user(user)
+    if not user.is_authenticated:
```

Comment:
Looking at the `login_user` doc, I wrote this wrong originally. If you do

```python
if not login_user(user):
  ...your code
```

that will handle any login failures from flask-login. I incorrectly assumed it would throw.